pynobo.py

```diff
@@ -849,6 +849,31 @@ async def async_update_zone(self, zone_id, name=None, week_profile_id=None, temp
 
         await self.async_send_command(command)
 
+    def update_week_profile(self, week_profile_id, name=None, profile=None):
+        self._create_task(self.async_update_week_profile(week_profile_id, name, profile))
+
+    async def async_update_week_profile(self, week_profile_id, name=None, profile=None):
+        """
+        Update the name and profile parameter for a week.
+
+        :param week_profile_id: the week_profile_id
+        :param name: the new zone name (default None)
+        :param profile: the new profile (default None)
+        """
+
+        # Initialize command with the current zone settings
+        command = [nobo.API.UPDATE_WEEK_PROFILE] + list(self.week_profiles[week_profile_id].values())
+
+        # Replace command with arguments that are not None. Is there a more elegant way?
+        if name:
+            command[2] = name
```

Comment:
Check that `name` does not contain illegal characters (e.g `,`)

Reply:
I can do this, but in the other function is no checking too. So is this library responsible for the valid data transfer or the upper layer or the "hardware" to send a error+ handling this? What is the design idea?

Reply:
Good question! I think all the methods that sends commands to the hub should to this, to avoid sending illegal messages and in worst case crash or brick the hub.

I'll look into this for existing methods.

Reply:
I have started on this in PR #32

It actually looks like the only requirement is to translate space to non breaking space, and limit the week profile name to 150 bytes (when it's UTF-8 encoded). E.g.:
```
name = name.replace(" ", "\u00A0")
if len(name.encode('utf-8')) > 150:
    raise ValueError(f'Week profile name "{name}" too long (max 150 bytes when encoded as UTF-8)')
```

Reply:
I don't want to collide I will merge your commit to my code

Reply:
It's merged to master now, so you can just rebase.

Reply:
@HelgeBrands If you rebase, I'm happy to review again.

Reply:
sorry i'm a little bit busy, please give me some time